### PR TITLE
CDAP-13634 use AppSpecAdapter to correctly serialize AppSpec

### PR DESCRIPTION
**Pull request by yaojiefeng**
JIRA: https://issues.cask.co/browse/CDAP-13634
Build: https://builds.cask.co/browse/CDAP-DUT6522-3

The app specification of the tracker app has some schema in it and we didn't use the correct Gson to serialize it. When we deserialize it, we get NPE for some schema fields. We need to use the correct Adaptor to serialize the app spec

### CODE REVIEW

**Comment by rohitsinha54** [comment]
Can you add unit test for AdminEventPublisher for the schema on which it failed?

**Review by albertshau — APPROVED** [review]
lgtm

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataSubscriberServiceTest.java [hunk 1]
```diff
@@ -439,6 +448,56 @@ public void testProfileMetadataWithNoProfilePreferences() throws Exception {
     }
   }
 
+  @Test
+  public void testAppDeletionMessage() throws Exception {
```
Comment (rohitsinha54) [inline cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataSubscriberServiceTest.java:452]: Does this test fail without the above changes ?

Comment (yaojiefeng) [reply cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataSubscriberServiceTest.java:452]: Yes, if you use new Gson() in the AdminEventPublisher, the test will fail

**Comment by rohitsinha54** [comment]
LGTM

**Comment by rohitsinha54** [comment]
Please close https://github.com/caskdata/cdap/pull/10268 after merging this.

**Pull request merged**